airflow/executors/kubernetes_executor_utils.py

```diff
@@ -143,6 +146,13 @@ def _run(
             self.log.debug("Event: %s had an event of type %s", task.metadata.name, event["type"])
             if event["type"] == "ERROR":
                 return self.process_error(event)
+            # Schedulers are issuing abrupt pod deletes when there is a delay in schedulers' heartbeat
+            # https://github.com/apache/airflow/issues/31198
+            # Added below scheduler_job_id condition to skip the events of pods created by other schedulers
```

Comment:
I would move these comments (here and on 132) to (1) the commit message and (2) the test.  Don't think they are needed here.